crates/red_knot/src/types/infer.rs

```diff
@@ -237,67 +238,74 @@ mod tests {
         Ok(TestCase { temp_dir, db, src })
     }
 
-    #[test]
-    fn follow_import_to_class() -> anyhow::Result<()> {
-        let case = create_test()?;
-        let db = &case.db;
+    fn write_to_path(case: &TestCase, relpath: &str, contents: &str) -> anyhow::Result<()> {
+        let path = case.src.path().join(relpath);
+        std::fs::write(path, dedent(contents))?;
+        Ok(())
+    }
 
-        let a_path = case.src.path().join("a.py");
-        let b_path = case.src.path().join("b.py");
-        std::fs::write(a_path, "from b import C as D; E = D")?;
-        std::fs::write(b_path, "class C: pass")?;
-        let a_file = resolve_module(db, ModuleName::new("a"))?
+    fn get_public_type(case: &TestCase, modname: &str, varname: &str) -> anyhow::Result<Type> {
+        let db = &case.db;
+        let file = resolve_module(db, ModuleName::new(modname))?
             .expect("module should be found")
             .path(db)?
             .file();
-        let a_syms = symbol_table(db, a_file)?;
-        let e_sym = a_syms
-            .root_symbol_id_by_name("E")
-            .expect("E symbol should be found");
+        let syms = symbol_table(db, file)?;
+        let sym = syms
+            .root_symbol_id_by_name(varname)
+            .expect("symbol should be found");
 
-        let ty = infer_symbol_public_type(
+        Ok(infer_symbol_public_type(
             db,
             GlobalSymbolId {
-                file_id: a_file,
-                symbol_id: e_sym,
+                file_id: file,
+                symbol_id: sym,
             },
-        )?;
+        )?)
+    }
 
-        let jar = HasJar::<SemanticJar>::jar(db)?;
-        assert!(matches!(ty, Type::Class(_)));
-        assert_eq!(format!("{}", ty.display(&jar.type_store)), "Literal[C]");
+    fn assert_public_type(
+        case: &TestCase,
+        modname: &str,
+        varname: &str,
+        tyname: &str,
+    ) -> anyhow::Result<()> {
+        let ty = get_public_type(case, modname, varname)?;
 
+        let jar = HasJar::<SemanticJar>::jar(&case.db)?;
+        assert_eq!(format!("{}", ty.display(&jar.type_store)), tyname);
```

Comment:
It feels kinda unfortunate that we have to convert the type to a string in order to assert that it is what we expect. But I agree that this is for the best, as it's going to allow us to write much more readable tests.

Reply:
On the other hand, we may eventually want to write tests based on `reveal_type` or `assert_type` anyway, which also use the textual representation of a type.